fem/bilininteg.cpp

```diff
@@ -832,15 +852,31 @@ void DiffusionIntegrator::AssembleElementMatrix
 
    const IntegrationRule *ir = IntRule ? IntRule : &GetRule(el, el);
 
+   const NURBSFiniteElement *NURBSFE =
+      dynamic_cast<const NURBSFiniteElement *>(&el);
+
+   IntegrationRule *irn = nullptr;
```

Comment:
Why not just use `ir` and avoid the two checks for `irn` below?

Reply:
It would be better to use only one `IntegrationRule*`, but `ir` has to be `const`. I thought `ir` could not be changed. Since you mentioned this, I realize it can be changed (because the `const` is before the type). This C++ feature can be confusing.